zerver/views/realm.py

```diff
@@ -60,139 +73,111 @@ def parse_jitsi_server_url(
-    ),
-    require_unique_names: Optional[bool] = REQ(json_validator=check_bool, default=None),
-    name_changes_disabled: Optional[bool] = REQ(json_validator=check_bool, default=None),
-    email_changes_disabled: Optional[bool] = REQ(json_validator=check_bool, default=None),
-    avatar_changes_disabled: Optional[bool] = REQ(json_validator=check_bool, default=None),
-    inline_image_preview: Optional[bool] = REQ(json_validator=check_bool, default=None),
-    inline_url_embed_preview: Optional[bool] = REQ(json_validator=check_bool, default=None),
-    add_custom_emoji_policy: Optional[int] = REQ(
-        json_validator=check_int_in(Realm.COMMON_POLICY_TYPES), default=None
-    ),
-    delete_own_message_policy: Optional[int] = REQ(
-        json_validator=check_int_in(Realm.COMMON_MESSAGE_POLICY_TYPES), default=None
-    ),
-    message_content_delete_limit_seconds_raw: Optional[Union[int, str]] = REQ(
-        "message_content_delete_limit_seconds", json_validator=check_string_or_int, default=None
-    ),
-    allow_message_editing: Optional[bool] = REQ(json_validator=check_bool, default=None),
-    edit_topic_policy: Optional[int] = REQ(
-        json_validator=check_int_in(Realm.EDIT_TOPIC_POLICY_TYPES), default=None
-    ),
-    mandatory_topics: Optional[bool] = REQ(json_validator=check_bool, default=None),
-    message_content_edit_limit_seconds_raw: Optional[Union[int, str]] = REQ(
-        "message_content_edit_limit_seconds", json_validator=check_string_or_int, default=None
-    ),
-    allow_edit_history: Optional[bool] = REQ(json_validator=check_bool, default=None),
-    default_language: Optional[str] = REQ(default=None),
-    waiting_period_threshold: Optional[int] = REQ(converter=to_non_negative_int, default=None),
-    authentication_methods: Optional[Dict[str, Any]] = REQ(
-        json_validator=check_dict([]), default=None
-    ),
+    *,
+    name: Annotated[
+        Optional[str], StringConstraints(max_length=Realm.MAX_REALM_NAME_LENGTH)
+    ] = None,
+    description: Annotated[
+        Optional[str], StringConstraints(max_length=Realm.MAX_REALM_DESCRIPTION_LENGTH)
+    ] = None,
+    emails_restricted_to_domains: Optional[Json[bool]] = None,
+    disallow_disposable_email_addresses: Optional[Json[bool]] = None,
+    invite_required: Optional[Json[bool]] = None,
+    invite_to_realm_policy: Optional[Json[InviteToRealmPolicyEnum]] = None,
+    create_multiuse_invite_group_id: Annotated[
+        Optional[Json[int]], ApiParamConfig(whence="create_multiuse_invite_group")
+    ] = None,
+    require_unique_names: Optional[Json[bool]] = None,
+    name_changes_disabled: Optional[Json[bool]] = None,
+    email_changes_disabled: Optional[Json[bool]] = None,
+    avatar_changes_disabled: Optional[Json[bool]] = None,
+    inline_image_preview: Optional[Json[bool]] = None,
+    inline_url_embed_preview: Optional[Json[bool]] = None,
+    add_custom_emoji_policy: Optional[Json[CommonPolicyEnum]] = None,
```

Comment:
One another way here could be to `check_int_in` with `AfterValidator`

Reply:
I think it's nice to get a `CommonPolicyEnum` out of it at the end.